lib/IDE/CompletionLookup.cpp

```diff
@@ -476,7 +476,11 @@ void CompletionLookup::addValueBaseName(CodeCompletionResultBuilder &Builder,
                                         DeclBaseName Name) {
   auto NameStr = Name.userFacingName();
   bool shouldEscapeKeywords;
-  if (Name.isSpecial()) {
+  if (Name.mustAlwaysBeEscaped()) {
+    // Names that are raw identifiers must always be escaped regardless of
+    // their position.
+    shouldEscapeKeywords = true;
```

Comment:
This calls `Lexer:: identifierMustAlwaysBeEscaped(StringRef)` twice, in `Name.mustAlwaysBeEscaped()` and `Builder.escapeKeyword()` which is a waste. Can we avoid that? I'm fine with just manually wrapping it with back-ticks here, or you could factor out the escaping logic from  `Builder.escapeKeyword`.

Reply:
Done. I've factored out a `escapeWithBackticks` method in the builder so it can be called directly.